src/SFML/Window/Window.cpp

```diff
@@ -61,6 +61,17 @@ Window::~Window()
 }
 
 
+////////////////////////////////////////////////////////////
+Window::Window(Window&& other) noexcept
+{
+    *this = std::move(other);
+}
```

Comment:
Why isn't this defaulted?

Reply:
the default move constructor was also deleting `shareContext` after goes out of scope. for testing when the previous test goes out of scope the next window creating fail assert finding `shareContext` on creating new window.  
you can test the default move constructor with the test below. the second `movedWindow` find `shareContext` nullptr while creating `m_context`.   
It seems the default move constructor gives the ownership of the `shareContext` to the `sf::Window` class and it gets deleted after goes out of scope.  
  
```c++
SUBCASE("Construction")
{
    sf::Window       movedWindow(sf::VideoMode(sf::Vector2u(256, 256), 32), "Window Title");
    const sf::Window window(std::move(movedWindow));
    CHECK(window.getSize() == sf::Vector2u(256, 256));
}

SUBCASE("Assignment")
{
    sf::Window movedWindow(sf::VideoMode(sf::Vector2u(256, 256), 32), "Window Title");
    sf::Window window;
    window = std::move(movedWindow);
    CHECK(window.getSize() == sf::Vector2u(256, 256));
}
```

Reply:
Couldn't it be cleaner by making GlResource class movable?

Reply:
I always prefer making base classes moveable first if possible since you can end up with weird implementations if base classes or class members are not moveable.

Reply:
thanks, I did that.
the problem was `resourceCount` was going to negative since the moved object was still exist and call destructor.